rbac/management/utils.py

```diff
@@ -65,24 +67,48 @@ def get_principal_from_request(request):
     return get_principal(username, request, verify_principal=bool(qs_user), from_query=from_query)
 
 
-def get_principal(username, request, verify_principal=True, from_query=False):
+def get_principal(
+    username: str, request: Request, verify_principal: bool = True, from_query: bool = False
+) -> Principal:
     """Get principals from username."""
     # First check if principal exist on our side,
     # if not call BOP to check if user exist in the account.
-    tenant = request.tenant
+    tenant: Tenant = request.tenant
     try:
-        # If the username was provided through a query we must verify if it is an org admin from the BOP
+        # If the username was provided through a query we must verify if it exists in the corresponding services first.
         if from_query:
-            verify_principal_with_proxy(username, request, verify_principal=verify_principal)
+            if ITService.is_username_service_account(username=username):
+                it_service = ITService()
+                if not it_service.is_service_account_valid_by_username(
+                    user=request.user, service_account_username=username
+                ):
+                    raise serializers.ValidationError(
+                        {"detail": f"No data found for service account with username '{username}'"}
+                    )
+            else:
+                verify_principal_with_proxy(username, request, verify_principal=verify_principal)
         principal = Principal.objects.get(username__iexact=username, tenant=tenant)
     except Principal.DoesNotExist:
-        verify_principal_with_proxy(username, request, verify_principal=verify_principal)
+        # If the "from query" parameter was specified, the username was validated above, so there is no need to
+        # validate it again.
+        if not from_query:
+            if ITService.is_username_service_account(username):
+                it_service = ITService()
+                if not it_service.is_service_account_valid_by_username(
+                    user=request.user, service_account_username=username
+                ):
+                    raise serializers.ValidationError(
+                        {"detail": f"No data found for service account with username '{username}'"}
+                    )
+            else:
+                verify_principal_with_proxy(username, request, verify_principal=verify_principal)
 
         # In the case that the user that made the request was a service account, store it accordingly in the database.
```

Comment:
Seems we can remove this comment, which seems not valid anymore

Reply:
Removed!